ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -706,10 +709,27 @@ private Page<NamedBlobRecord> run_list_v2(String accountName, String containerNa
         int resultIndex = 0;
         while (resultSet.next()) {
           String blobName = resultSet.getString(1);
-          if (resultIndex++ == maxKeysValue) {
+          if (resultIndex == maxKeysValue) {
```

Comment:
we only set nextContinuationToken to the blobName when resultIndex == maxKeysValue. The idea is to only do that when it's the last key, but now that we change the logic when incrementing resultIndex. resultIndex would not equal to maxKeysValue in the last key, so we probably have to use an other index, or other way to set next continuation token here.

Reply:
Sure. Reverted back to previous logic of incrementing `resultIndex` on every iteration and  setting `nextContinuationToken` to the blobName when `resultIndex == maxKeysValue`